application/browser/linux/running_application_object.cc

```diff
@@ -319,7 +319,8 @@ void RunningApplicationObject::SendChannel(
   dbus::MessageWriter writer(response.get());
   writer.AppendString(ep_bp_channel_.name);
 
-  scoped_ptr<dbus::FileDescriptor> client_fd(new dbus::FileDescriptor(fd));
+  scoped_ptr<dbus::FileDescriptor> client_fd(new dbus::FileDescriptor());
+  client_fd->PutValue(fd);
```

Comment:
Passing the fd in the constructor doesn't transfer ownership but PutValue() will do it.
When dbus::FileDescriptor goes out of scope it will close the fd.